pkg/gofr/database_test.go

```diff
@@ -0,0 +1,29 @@
+package gofr
+
+import (
+	"database/sql"
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_newMySQL(t *testing.T) {
+	testCases := []struct {
+		desc   string
+		port   string
+		expDB  *sql.DB
+		expErr error
+	}{
+		{"db connected successfully", "3306", &sql.DB{}, nil},
+		{"db connection  failed", "2001", nil, &net.OpError{}},
+	}
+
+	for _, tc := range testCases {
```

Comment:
we can add iterator and use it in test failure msg to specify which test failed.